mipengine/controller/algorithm_executor.py

```diff
@@ -361,12 +231,230 @@ def get_run_udf_query(
             positional_args=positional_args,
         )
 
-    # CLEANUP functionality
-
     def clean_up(self):
         self._node_tasks_handler.clean_up(context_id=self.context_id)
 
 
+class INodeTable(ABC):
+    # TODO: better abstraction here...
+    pass
+
+
+class _LocalNodeTable(INodeTable):
+    def __init__(self, nodes_tables: Dict[_Node, TableName]):
+        self._nodes_tables = nodes_tables
+
+        if not self._validate_matching_table_names(list(self._nodes_tables.values())):
```

Comment:
Why not throw the exception inside the `validate` method?

Reply:
fixed